File: api/src/main/java/cender/shop/PL/dto/ProductDto.java
package cender.shop.PL.dto;

import lombok.AllArgsConstructor;
import lombok.Data;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

@Data
public class ProductDto {

    @NotBlank
    @NotNull
    public String title;
    @NotBlank
    @NotNull
    public String vendor;
    @NotBlank
    @NotNull
    public String rating;
    @NotBlank
    @NotNull
    public String productDescription;

    public float price;

    public ProductDto(String title, String vendor, String rating, String productDescription, float price){
        this.title = title;
        this.vendor = vendor;
        this.rating = rating;
        this.productDescription = productDescription;
        this.price = price;
    }
}

File: react/src/types/dtos/product/productDto.ts
export interface productDto {
  title: string;
  vendor: {
    id: number;
    vendorName: string;
  };
  rating: string;
  productDescription: string;
  price: number;
}

export interface updateProductDto extends productDto {
  id: number;
}

export interface deleteProduct {
  id: number;
}
